Sources/GDSAnalytics/Screens/ScreenView.swift

```diff
@@ -2,22 +2,36 @@ public protocol ScreenViewProtocol {
     associatedtype Screen: NamedScreen
     var screen: Screen { get }
     var title: String { get }
+    var type: String? { get }
+    var screenID: String? { get }
     var parameters: [String: String] { get }
 }
 
 public struct ScreenView<Screen: NamedScreen>: ScreenViewProtocol {
     public let screen: Screen
     public let title: String
+    public var type: String?
+    public let screenID: String?
     
     public var parameters: [String: String] {
-        [ScreenParameter.title.rawValue: title]
-            .mapValues(\.formattedAsParameter)
+        if let screenID, let type {
+            return         ["screen_id": screenID,
+                            "type": type]
+                               .mapValues(\.formattedAsParameter)
+        } else {
+            return [ScreenParameter.title.rawValue: title]
+                .mapValues(\.formattedAsParameter)
+        }
```

Comment:
```suggestion
        [
            ScreenParameter.id.rawValue: screenID,
            ScreenParameter.title.rawValue: title,
            ScreenParameter.type.rawValue: type,
        ].compactMapValues(\.formattedAsParameter)
```

Reply:
<img width="715" alt="image" src="https://github.com/govuk-one-login/mobile-ios-common/assets/39125995/175a3960-4b95-43d4-9222-0dec5488dce8">
Just checking, does this change replace the if let or just make it nicer to look at? Doing it this way still results in the optional values being an issue :(